### PR TITLE
Upgrade application to rails 5

### PR DESCRIPTION
**Pull request by Seunadex**
#### Is this solving an issue already in our backlog? If so, please mention which one
[588](https://github.com/rubycorns/rorganize.it/issues/588)

#### Special things about this PR to be considered
This PR upgrades the application  to rails 5.0.5, a lot files were changed so the application might need intensive testing and code review to be 100% sure everything works fine

#### Do you have any open Questions you need to get answered?
The current version is `5.0.5` , is that cool or do we still need to go a bit higher?

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -1,29 +1,34 @@
 #!/usr/bin/env ruby
 require 'pathname'
+require 'fileutils'
+include FileUtils
```
Comment (PragTob) [inline bin/setup]: is `include FileUtils` needed here? Seems weird, we're not in any class so what would it do? 🤔

File: bin/update [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
+require 'pathname'
+require 'fileutils'
+include FileUtils
```
Comment (PragTob) [inline bin/update]: same as above?

File: config/environments/production.rb [hunk 3]
```diff
@@ -14,13 +14,9 @@
...
-  # Enable Rack::Cache to put a simple HTTP cache in front of your application
-  # Add `rack-cache` to your Gemfile before enabling this.
-  # For large-scale production use, consider using a caching reverse proxy like nginx, varnish or squid.
-  # config.action_dispatch.rack_cache = true
-
-  # Disable Rails's static asset server (Apache or nginx will already do this).
-  config.serve_static_assets = true
+  # Disable serving static files from the `/public` folder by default since
+  # Apache or NGINX already handles this.
+  config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (PragTob) [inline config/environments/production.rb]: not sure we set or don't set tis

File: config/environments/production.rb [hunk 4]
```diff
@@ -29,37 +25,37 @@
...
+  # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
-  # Set to :debug to see everything in the log.
-  config.log_level = :info
+  # Use the lowest log level to ensure availability of diagnostic information
+  # when problems arise.
+  config.log_level = :debug
```
Comment (PragTob) [inline config/environments/production.rb]: log level of debug seems wrong for prod

File: config/environments/production.rb [hunk 5]
```diff
@@ -88,9 +84,19 @@
...
-  # config.autoflush_log = false
-
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
+
+  # Use a different logger for distributed setups.
+  # require 'syslog/logger'
+  # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
+
+  if ENV["RAILS_LOG_TO_STDOUT"].present?
```
Comment (PragTob) [inline config/environments/production.rb]: not sure if we have this set again

File: spec/controllers/admin_controller_spec.rb [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (PragTob) [inline spec/controllers/admin_controller_spec.rb]: this seems... odd?

File: spec/controllers/registrations_controller_spec.rb [hunk 7]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (PragTob) [inline spec/controllers/registrations_controller_spec.rb]: all of thsoe seem odd

File: config/puma.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# Puma can serve each request in a thread from an internal thread pool.
+# The `threads` method setting takes two numbers a minimum and maximum.
+# Any libraries that use thread pools should be configured to match
+# the maximum value specified for Puma. Default is set to 5 threads for minimum
+# and maximum, this matches the default thread size of Active Record.
+#
```
Comment (PragTob) [inline config/puma.rb]: would we be running puma now? Nothing against puma but we might need to get rid off thin ten and be careful

File: spec/factories/groups.rb [hunk 9]
```diff
@@ -25,7 +25,7 @@
 #  street     :string(255)
 #
 
-FactoryGirl.define do
+FactoryBot.define do
```
Comment (PragTob) [inline spec/factories/groups.rb]: 👍

File: spec/models/person_spec.rb [hunk 10]
```diff
@@ -25,243 +25,247 @@
...
+require 'rails_helper'
 
-  it { is_expected.to have_many(:groups) }
-  it { is_expected.to have_many(:memberships) }
-  it { is_expected.to have_many(:topics) }
-  it { is_expected.to have_many(:posts) }
+RSpec.describe Person, type: :model do
 
-  it { is_expected.to validate_presence_of(:first_name) }
+  describe Person, vcr: {cassette_name: 'create_group'} do
```
Comment (PragTob) [inline spec/models/person_spec.rb]: I don't think we need this second describe block?

File: bin/setup (hunk 1 above)
Comment (Seunadex) [reply bin/setup]: @PragTob, Actually this was auto generated

File: config/puma.rb (hunk 8 above)
Comment (Seunadex) [reply config/puma.rb]: @PragTob kindly elaborate on this.

**Comment by Seunadex** [comment]
@PragTob All comments has been addressed

**Comment by vsmart** [comment]
:wave: I tried to checkout your branch locally for testing and got this error every time:

 ```
DEPRECATION WARNING: You didn't set `secret_key_base`. Read the upgrade documentation to learn more about this new config option. (called from require at bin/rails:4)
Unexpected error while processing request: Missing `secret_key_base` for 'development' environment, set this value in `config/secrets.yml`
	/home/franka/.gem/ruby/2.4.3/gems/railties-5.0.5/lib/rails/application.rb:513:in `validate_secret_key_config!'
	/home/franka/.gem/ruby/2.4.3/gems/railties-5.0.5/lib/rails/application.rb:246:in `env_config'
	/home/franka/.gem/ruby/2.4.3/gems/railties-5.0.5/lib/rails/engine.rb:693:in `build_request'
	/home/franka/.gem/ruby/2.4.3/gems/railties-5.0.5/lib/rails/application.rb:521:in `build_request'
	/home/franka/.gem/ruby/2.4.3/gems/railties-5.0.5/lib/rails/engine.rb:521:in `call'
	/home/franka/.gem/ruby/2.4.3/gems/thin-1.7.2/lib/thin/connection.rb:86:in `block in pre_process'
	/home/franka/.gem/ruby/2.4.3/gems/thin-1.7.2/lib/thin/connection.rb:84:in `catch'
	/home/franka/.gem/ruby/2.4.3/gems/thin-1.7.2/lib/thin/connection.rb:84:in `pre_process'
```

Looks like the `secret-key-base` key is missing, perhaps there was a change here in Rails 5. Do you happen to know what the issue may be?

**Comment by Seunadex** [comment]
@vsmart  Check if you have the `secret-key-base` set in your `config/secrets.yml` file for development environment

**Comment by vsmart** [comment]
@Seunadex Ah yes. I forgot to do this step while cloning your repo fork. 
I clicked around a bit and it looks good :ok_hand:

**Comment by lislis** [comment]
yeesssss

**Pull request merged**